File: liemipeli/src/main/java/com/antero/liemipeli/domain/opiskelu/Koulu.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.antero.liemipeli.domain.opiskelu;

import com.antero.liemipeli.domain.opiskelu.Kurssitoteutus;
import java.util.ArrayList;
import java.util.Collection;

/**
 *
 * @author Antero Oikkonen
 */
public class Koulu {
    private String nimi;
    private Collection<Opettaja> opettaja;
    private Collection<Kurssitoteutus> kurssiToteutukset;
    private int lukukausimaksu;

    public String getNimi() {
        return nimi;
    }

    public void setNimi(String nimi) {
        this.nimi = nimi;
    }

    public Collection<Opettaja> getOpettaja() {
        return opettaja;
    }

    public void setOpettaja(Collection<Opettaja> opettaja) {
        this.opettaja = opettaja;
    }

    public Collection<Kurssitoteutus> getKurssiToteutukset() {
        return kurssiToteutukset;
    }

    public void setKurssiToteutukset(Collection<Kurssitoteutus> kurssiToteutukset) {
        this.kurssiToteutukset = kurssiToteutukset;
    }

    public int getLukukausimaksu() {
        return lukukausimaksu;
    }

    public void setLukukausimaksu(int lukukausimaksu) {
        this.lukukausimaksu = lukukausimaksu;
    }

    /**
     * lisätään <code>Kurssitoteutus</code>
     * @param ktot 
     */
    
    public void addKurssitoteutus(Kurssitoteutus ktot) {
       if(this.kurssiToteutukset==null){
           kurssiToteutukset = new ArrayList<Kurssitoteutus>();
       }
       kurssiToteutukset.add(ktot);
    }
    
    
    
    
}

File: liemipeli-kotlin/src/main/kotlin/com/antero/liemipeli/opiskelu/Koulu.kt
package com.antero.liemipeli.opiskelu

import java.util.ArrayList



/**
 *
 * @author Antero Oikkonen
 */
class Koulu {

    var nimi: String? = null
    var opettaja: Collection<Opettaja>? = null
    private var kurssiToteutukset: MutableCollection<Kurssitoteutus>? = null
    var lukukausimaksu: Int = 0

    fun getKurssiToteutukset(): Collection<Kurssitoteutus>? {
        return kurssiToteutukset
    }

    fun setKurssiToteutukset(kurssiToteutukset: MutableCollection<Kurssitoteutus>) {
        this.kurssiToteutukset = kurssiToteutukset
    }

    /**
     * lisätään `Kurssitoteutus`
     *
     * @param ktot
     */
    fun addKurssitoteutus(ktot: Kurssitoteutus) {
        if (this.kurssiToteutukset == null) {
            kurssiToteutukset = ArrayList()
        }
        kurssiToteutukset!!.add(ktot)
    }

    override fun toString(): String {
        return "$nimi,$opettaja, lukukausimaksu $lukukausimaksu"
    }
}
